examples/tendermint.rs

```diff
@@ -0,0 +1,230 @@
+        info!("Sql query: {}", sql);
+
+        let dialect = GenericDialect {};
+        let mut resp = ResponseCheckTx::new();
+        // TODO: cover sql injection
+        match Parser::parse_sql(&dialect, sql.clone()) {
+            Ok(_val) => {
+                info!("Valid SQL");
+                // Update state to keep state correct for next check_tx call
+                self.sql = sql;
+            }
+            Err(_e) => {
+                info!("Invalid SQL");
+                // Return error
+                resp.set_code(1);
+                resp.set_log(String::from("Must be valid sql!"));
+            }
+        }
+        return resp;
+    }
+
+    fn deliver_tx(&mut self, req: &RequestDeliverTx) -> ResponseDeliverTx {
+        info!("deliver_tx()");
+
+        // Get the Tx [u8]
+        let sql = convert_tx(req.get_tx());
+        // Update state
+        self.sql = sql;
+        // Return default code 0 == bueno
+        ResponseDeliverTx::new()
+    }
+
+    fn commit(&mut self, _req: &RequestCommit) -> ResponseCommit {
+        info!("commit() {}", self.sql);
+
+        // Run Query
+        //Runtime::new().unwrap().block_on(run_query_async(&self.sql));
+        run_query_sync(self.sql.clone());
+
+        // Create the response
+        let mut resp = ResponseCommit::new();
+
+        // Set data so last state is included in the block
+        let bytes = self.sql.as_bytes();
+        resp.set_data(bytes.to_vec());
+
+        self.sql = String::from("");
+        resp
+    }
+}
+
+struct ProxyHandler {
+    http_client: Client<HttpConnector, Body>,
+}
+
+// Just forward the packet
+impl PacketHandler for ProxyHandler {
+    fn handle_request(&mut self, p: &Packet) -> Packet {
+        // Print out the packet
+        //debug!("[{}]", String::from_utf8_lossy(&p.bytes));
+
+        match p.packet_type() {
+            // Forward all SQL queries to Tendermint
+            Ok(PacketType::ComQuery) => {
+                let payload = &p.bytes[5..];
+                let sql = String::from_utf8(payload.to_vec()).expect("Invalid UTF-8");
+                info!("SQL: {}", sql);
+                let mut url: String = "http://localhost:26657/broadcast_tx_commit?tx=".to_owned();
+                url.push_str(&sql);
+                info!("Pushing to Tendermint: {}", url);
+                let _fut = self.http_client.get(url.parse().unwrap()).then(|res| {
+                    async move {
+                        let response = res.unwrap();
+                        debug!("Response: {}", response.status());
+                        debug!("Headers: {:#?}\n", response.headers());
+                    }
+                });
+            }
+            _ => debug!("{:?} packet", p.packet_type()),
+        }
+
+        Packet {
+            bytes: p.bytes.clone(),
+        }
+    }
+
+    fn handle_response(&mut self, p: &Packet) -> Packet {
+        Packet {
+            bytes: p.bytes.clone(),
+        }
```

Comment:
```rust
p.clone()
```
(per your `derive(Clone)` in the other branch

Reply:
yep. i added that later, after this branch was made.